icees_api/features/sql.py

```diff
@@ -510,11 +510,9 @@ def count_unique(conn, table_name, *columns):
         [2, 2, 1]
     ]
     """
+    scolumns = list(map(column, columns))
     return [list(row) for row in conn.execute(
-        "SELECT {cols}, count(*) FROM {table_name} GROUP BY {cols}".format(
-            cols=", ".join(f"\"{col}\"" for col in columns),
-            table_name=table_name,
-        )
+        select(scolumns + [func.count()]).select_from(table(table_name)).group_by(*scolumns)
```

Comment:
I've been deliberately trying to remove some of the SQLAlchemy in favor of generating SQL natively. This is partly just because I prefer reading/writing SQL to the SQLAlchemy API, but also it's contributing to some quite egregious computational inefficiencies in our SQL queries. We feel them of course, but don't see them because the actual SQL is hidden behind the SQLAlchemy API. I would like to ultimately remove SQLAlchemy entirely. I don't think SQLAlchemy is a very good fit for our situation. It works better when dealing with known data structures/database schemas, but we're dynamically generating the SQLAlchemy structures and then using them to generate the SQL... It feels unnecessary to me and harder to read than if we just generated the SQL directly. One advantage of SQLAlchemy that @xu-hao mentioned is the ability to smooth over the differences between engines (SQLite and PostgreSQL in our case). This is valid, I'm just not sure it outweighs the cons in our situation.

Reply:
we're trying to avoid sql injection

Reply:
That's a good point to consider. I think we should plan to validate user inputs early on, both to check them against an allowlist, preventing injection attacks, and to provide useful feedback re: invalid table/column names. It's not clear to me from the documentation that using `table()` here will prevent injection attacks if `table_name` is nasty.

Reply:
See #134.